### ISSUE TITLE
Request help on the weight regularization loss in detectron2

### ISSUE THREAD

**Issue by Remember2018** [facebookresearch/detectron2#239]
Hi, I have a question on the weight regularization loss in `detectron2`. In the line 
https://github.com/facebookresearch/detectron2/blob/046f035d7e2223a8eefe3f5f29880171580fe856/detectron2/engine/train_loop.py#L212

the losses is from the model outputs (i.e. the losses function of the model), but in terms of some archtectures, e.g. the retinanet
https://github.com/facebookresearch/detectron2/blob/046f035d7e2223a8eefe3f5f29880171580fe856/detectron2/modeling/meta_arch/retinanet.py#L197

the losses didn't include the weight regularization loss. So in the framework of `detectron2`, the regularization losses are not included? But in the framework of `Tensorpack`, I found the weight decay losses are included, e.g., 

https://github.com/tensorpack/tensorpack/blob/23ab7001921304961027e9ad3e15261a85732c1d/examples/FasterRCNN/modeling/generalized_rcnn.py#L77

I'm not sure if I understood correctly. Could you please give some advice? Thanks.

**Comment by ppwwyyxx** [comment]
In pytorch, weight regularization is often handled by optimizer. And so we did in detectron2.

**Comment by Remember2018** [comment]
Hi @ppwwyyxx thank you Yuxin!
